docs/function_calling.rst

```diff
@@ -137,6 +137,10 @@ Next Actor
 After a function call returns, kani will hand control back to the LM to generate a response by default. If instead
 control should be given to the human (i.e. return from the chat round), set ``after=ChatRole.USER``.
 
+.. note::
+    If the model calls multiple tools in parallel, the model will be allowed to generate a response if *any* function
+    allows it.
```

Comment:
This is interesting behavior (and not one that I particularly expected). I kind of expected the behavior to be waiting on all functions to finish instead of returning if any finish. 

Does OpenAI allow us to specify that we want to wait on all functions? Am I misunderstanding this?

Reply:
I think you might be misunderstanding - this note is specifically for if multiple different `@ai_function`s with different `after=ChatRole.USER/ChatRole.ASSISTANT` are used in a single parallel tool call. It does wait for all to finish before returning control to the user/model

Reply:
ah got it, ty